docs/authentication.md

```diff
@@ -0,0 +1,211 @@
+# Authentication Functions
+
+<!-- Glee relies on functions to execute your business logic. Functions are files that export a default async Node.js function: -->
+
+Authentication in Glee can be done using authentication functions. Authentication functions are files that export either one or both of the `clientAuth` and `serverAuth` Node.js functions:
+
+```js
+/* websocket.js */
+
+export async function serverAuth({ authProps, done }) {
+  //server auth logic
+}
+
+export async function clientAuth({ parsedAsyncAPI, serverName }) {
+  //client auth logic
+}
+```
+
+Glee looks for authentication files in the `auth` directory, the name of the authentication file should be the name of the targeted server that the authentication logic should work for.
+
+## Supported Authentication Values in AsyncAPI.yaml file
+
+AsyncAPI currently supports a variety of authentication formats as specified in the documentation, however Glee supports the following authentication schemas.
+
+A sample `asyncAPI.yaml` for a server with security requirments and security schemes is shown below:
+
+```yaml
+##server asyncAPI schema
+asyncapi: 2.6.0
+info:
+  title: AsyncAPI IMDB server
+  version: 1.0.0
+  description: This app is a dummy server that would stream the trending/upcoming anime.
+servers:
+  trendingAnimeServer:
+    url: 'http://localhost:8081'
+    protocol: http
+    security:
+      - token: []
+      - userPass: []
+      - apiKey: []
+      - UserOrPassKey: []
+      - oauth: []
```

Comment:
For the sake of simplicity, It would be great to demonstrate only one here. and maybe a simple one. like `userPass`